src/subs.fs

```diff
@@ -0,0 +1,54 @@
+namespace Elmish
+
+open System
+
+/// SubId - ID for live subscriptions, alias for string
+type SubId = string
```

Comment:
We had a discussion around the implications of having globally-scoped ids under the direct control of a (sub)component in the application. 
As it is the guarantees we have so far in elmish that each child can be viewed and understood in isolation would no longer hold. Not only one would need to know what their component is doing, all other components and potentially libraries would be playing in the same Id space.
Maybe `map` should take the scope as additional parameter?
Maybe Id should be a member of `Sub`?

Reply:
ID scope parameter on `map` is a great idea. End users who have parent and child subs will quickly realize the need for this. For libraries it can be treated similar to model or msg: if the library needs its own subscriptions, it can prefix the child sub IDs to ensure no conflicts, otherwise pass it on.

I initially made `Sub` to include SubId. It's quite arguable, but I thought the record version was a bit verbose, and the tuple made it feel like the key/value that it is. Comparison below. Happy to change it.

```fsharp
// tuple version
let subscribe model =
    [ "subId1", Time.every model.Interval (Time.getNow Ticked) ]

// record version
let subscribe model =
    [ { SubId = "subId1" // line break for > 80 chars
        Start = Time.every model.Interval (Time.getNow Ticked) } ]
```
_Could provide a helper fn to create the record more succinctly. But it will require extra parens: `[ Sub.mk "subId1" (Time.every ...) ]`, or a pipe syntax `[ Time.every ... |> Sub.mk "subId1" ]`._

Reply:
I added `idPrefix` as a parameter to `Subs.map`.

On Sub ID. I started to add it to Sub, but it seems to be at the wrong level of abstraction. Either the user will construct the record with SubId and be forced into extra detail (the Subscribe fn). Or subscription provider unnecessarily pays the overhead of constructing it and passing thru a sub id they can't use.

I considered cases where the subscriptions might want to use the sub id. Key-based features that need unique names, such as JS var names or storage keys. Putting SubId on Sub won't work for that. Sub creation only captures the original id. The subscribe fn can't access the updated version that may have been prefixed since then.

This leads me to think that SubId should be a parameter to Subscribe to support key-based features: `{ Subscribe: SubId -> Dispatch<'msg> -> IDisposable }` And also that SubId should still be kept separate from the subscribe fn. (Remembering that Sub is a record only to provide a Subscribe label for easier comprehension.) Maybe SubId can be an SCU for easier user comprehension?

```fsharp
let subscribe model =
    [ SubId "a/b/c/d", Time.every 1000 (Time.getNow Ticked) ]
```

NB: SCUs are [terrible for perf](https://bartoszsypytkowski.com/writing-high-performance-f-code/#discriminatedunionsasvaluetypes) (bottom of section). Fable can erase them, but not dotnet AFAIK. Funny, but this would accomplish the same thing. `let SubId (s: string) = s`.

Reply:
Let me know if this needs further work. @et1975